### ISSUE TITLE
iroh status command

### ISSUE THREAD

**Issue by faassen** [n0-computer/iroh#271]
I think this is the most developed CLI command of the bunch, with behavior particular to Iroh. So the current behavior may be enough for iroh cloud. What about iroh one?

There is a technical issue blocking the progress of moving this into the API so the CLI can use it; `watch` returns a Stream which is an `impl Trait` and I'm struggling to get it to work inside of the API trait, so I need some help.

**Comment by faassen** [comment]
Conclusion from meeting: it will be extended with an additional option to show versions of the various underlying services.

**Comment by b5** [comment]
help text overview. This introduces a bunch of augmentation to the status command, which I think we need to work through & prepare sample output for. I'll follow up with example output so we have something concrete to scrutinize

```
iroh-status
Report current status of iroh

USAGE:
    iroh status [OPTIONS]

OPTIONS:
    -h, --help     Print help information
    -w, --watch    Poll process for changes


status reports the current operational setup of iroh. Use status as a go-to
command for understanding where iroh commands are being processed. different
ops configurations utilize different network and service implementations
under the hood, which can lead to varying performance characteristics.

Status reports connectivity, which is either offline or online:

  offline: iroh is not connected to any background process, all commands
           are one-off, any network connections are closed when a command
           completes. Some network duties may be delegated to remote hosts.

  online:  iroh has found a long-running process to issue commands to. Any
           comand issued will be deletegated to the long-running process as a
           remote procedure call

If iroh is online, status also reports the service configuration of the
long running process, including the health of the configured subsystem(s).
Possible configurations fall into two buckets:

  one:     Iroh is running with all services bundled into one single process,
           this setup is common in desktop enviornments.

  cloud:   Iroh is running with services split into separate processes, which
           are speaking to each other via remote procedure calls.

Use the --watch flag to continually poll for changes.

Status reports no metrics about the running system aside from current service
health. Instead all metrics are emitted through uniform tracing collection &
reporting, which is intended to be consumed by tools like prometheus and
grafana. For more info on metrics collection, see
https://iroh.computer/docs/metrics
```

**Comment by faassen** [comment]
Some feedback:

"one single progress" -> I think that's not quite correct, multiple processes exist but a single *binary*. Did I get that right?

The most obvious missing bit is that there's no flag described yet to obtain version information.

**Comment by b5** [comment]
closed by #360
